src/header/header.scss

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: $iui-xxs solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
+  > .iui-center,
+  > .iui-right {
+    display: flex;
+    align-items: center;
+    height: 100%;
+  }
+
+  > .iui-left {
+    flex: 2 1 33%;
+    @include iui-header-buttons;
+
+    > .iui-chevron {
+      @include iui-icons-small;
+      flex-shrink: 0;
+      margin: 0 $iui-xs;
+    }
+  }
+
+  > .iui-center {
+    justify-content: center;
+    flex: 1 1 auto;
+  }
+
+  > .iui-right {
+    flex: 2 1 33%;
+    justify-content: flex-end;
+
+    .iui-user-icon {
+      transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out;
+        
+      * {
+        transition: width $iui-speed-fast ease-out, height $iui-speed-fast ease-out, font-size $iui-speed-fast ease-out, line-height $iui-speed-fast ease-out, font-weight $iui-speed-fast ease-out;
+      }
+    }
+  }
+
+  .iui-button {
+    height: 100%;
+    border-radius: 0;
+  }
+
+  .iui-divider {
+    width: 1px;
+    height: $iui-baseline * 2;
+    @include themed {
+      background-color: t(iui-color-background-4);
+    }
+  }
+  
+  &.iui-slim,
+  &.iui-slim-on-hover-demo:hover {
+    height: round($iui-baseline * 2.5);
+
+    > .iui-left {
+      > .iui-application {
+        > .iui-label {
+          font-size: $iui-font-size;
+          margin-right: $iui-l;
+        }
+
+        > .iui-brand-logo > .iui-icon {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+      }
+
+      > .iui-header-button {
+        > .iui-thumbnail-area > .iui-thumbnail {
+          width: $iui-icons-default;
+          height: $iui-icons-default;
+        }
+
+        &.iui-detailed > .iui-label {
+          padding-top: $iui-baseline * 2;
+
+          > .iui-id {
+            opacity: 0;
+          }
+        }
+      }
+    }
+    
+    > .iui-right {
+      .iui-user-icon {
+        @include iui-user-icon;
```

Comment:
Probably not necessary to `@include iui-user-icon` as `.iui-user-icon` already includes it (otherwise it would simply not work in non slim mode.. Only the size should be enough.